agreement/pseudonode.go

```diff
@@ -266,7 +266,19 @@ func (n asyncPseudonode) makeProposals(round basics.Round, period period, accoun
 	deadline := time.Now().Add(AssemblyTime)
 	ve, err := n.factory.AssembleBlock(round, deadline)
 	if err != nil {
-		n.log.Errorf("pseudonode.makeProposals: could not generate a proposal for round %d: %v", round, err)
+		if err == ErrAssembleBlockRoundStale {
+			// check to see if we're truely stale.
+			ledgerNextRound := n.ledger.NextRound()
+			if ledgerNextRound == round {
```

Comment:
I would get rid this check and the `else` block, and keep the changes to `agreement` as small as possible (i.e. `if err != ErrAssembleBlockRoundStale`, do the log)

This more complicated checks can be moved into `pool.AssembleBlock`, where you can compare against something like `pool.ledger.Latest()`

Reply:
If not under the `agreement`, then the `pool` isn't the best either. What about the AlgorandFullNode.AssembleBlock ? that's there the three component are joined.

Reply:
That sounds like a good option to me -- as a general rule I just want to keep the agreement changes as small as possible so that we don't have to think about them (even though this one is probably benign).